lib/fizzy/execute.cpp

```diff
@@ -223,13 +224,17 @@ execution_result execute(Instance& instance, FuncIdx function, std::vector<uint6
     // TODO: preallocate fixed stack depth properly
     Stack<uint64_t> stack;
 
-    Stack<LabelContext> labels;
-
     bool trap = false;
 
     const Instr* pc = code.instructions.data();
     const uint8_t* immediates = code.immediates.data();
 
+    Stack<LabelContext> labels;
+    //    bool have_result =
+    //    !instance.module.typesec[instance.module.funcsec[function]].outputs.empty();
+    labels.emplace_back(LabelContext{
+        &code.instructions[code.instructions.size() - 1], nullptr, have_result, stack.size()});
```

Comment:
I don't like that bool is casted to size_t here.

This can be deduced from the type section too. I think there should be a PR which enforces the type section on execute.